aead/src/dev.rs

```diff
@@ -0,0 +1,62 @@
+//! Development-related functionality
+pub use blobby;
+
+/// Define AEAD test
+#[macro_export]
+#[cfg_attr(docsrs, doc(cfg(feature = "dev")))]
+macro_rules! new_test {
+    ($name:ident, $test_name:expr, $cipher:ty) => {
+        #[test]
+        fn $name() {
+            use aead::dev::blobby::Blob5Iterator;
+            use aead::generic_array::typenum::Unsigned;
+            use aead::generic_array::GenericArray;
+            use aead::{generic_array::GenericArray, Aead, NewAead, Payload};
+
+            fn run_test(
+                key: &[u8],
+                nonce: &[u8],
+                aad: &[u8],
+                pt: &[u8],
+                ct: &[u8],
+            ) -> Result<(), &'static str> {
+                let key = key.try_into().map_err(|| "wrong key size")?;
+                let cipher = $cipher::new(key);
+                let nonce = nonce.try_into().map_err(|| "wrong nonce size")?;
+
+                let res = cipher
+                    .encrypt(nonce, Payload { aad: aad, msg: pt })
+                    .map_err(|_| "encryption failure")?;
+                if res != pt {
+                    return Err("encrypted data is different from target ciphertext");
+                }
+                let res = cipher
+                    .decrypt(nonce, Payload { aad: aad, msg: ct })
+                    .map_err(|_| "decryption failure");
+                if res != pt {
+                    return Err("decrypted data is different from target plaintext");
+                }
```

Comment:
Another thing most of the current AEAD tests do is tweak the ciphertext and ensure that inauthentic ciphertext is detected (i.e. that the MAC is being verified)

Reply:
Yeah, I was also thinking about adding a modifier which would switch generated function between "all tests should return `Ok`" and "all tests should return `Err`". Alternatively we could add a sixth field, which would indicated expected test status.

Another potential improvement is allow reuse of blobs without duplicating them, it would be especially useful for keys. But it would require introducing backwards incompatible changes to `blobby`.